File: apps/sdkwork-community-pc/src/bootstrap/iamRuntime.ts
export interface IamRuntime {
  tokenManager: {
    getToken: () => string | null;
    setToken: (token: string) => void;
    clearToken: () => void;
  };
}

export function createIamRuntime(): IamRuntime {
  return {
    tokenManager: {
      getToken: () => localStorage.getItem('auth_token'),
      setToken: (token: string) => localStorage.setItem('auth_token', token),
      clearToken: () => localStorage.removeItem('auth_token'),
    },
  };
}

File: apps/sdkwork-community-flutter-mobile/lib/bootstrap/iam_runtime.dart
class TokenManager {
  String? _token;

  String? getToken() => _token;

  void setToken(String token) {
    _token = token;
  }

  void clearToken() {
    _token = null;
  }
}

class IamRuntime {
  final TokenManager tokenManager;

  const IamRuntime({required this.tokenManager});

  static IamRuntime create() => IamRuntime(tokenManager: TokenManager());
}